packages/webcomponents/src/components/payment-details/payment-details-core.tsx

```diff
@@ -0,0 +1,111 @@
+import { Component, Host, h, Prop, State, Watch } from '@stencil/core';
+import { Payment } from '../../api';
+import { MapPaymentStatusToBadge, formatCurrency, formatDate, formatTime, snakeCaseToHumanReadable } from '../../utils/utils';
+import { CodeBlock, DetailItem, DetailSection, EntityHeadInfo, EntityHeadInfoItem, ErrorState, LoadingState } from '../details/utils';
+
+@Component({
+  tag: 'payment-details-core',
+  styleUrl: 'payment-details.scss',
+})
+
+export class PaymentDetailsCore {
+  @Prop() getPaymentDetails: Function;
+  @State() payment: Payment;
+  @State() loading: boolean = true;
+  @State() errorMessage: string;
+
+  componentWillLoad() {
+    if (typeof this.getPaymentDetails === 'function') {
+      this.fetchData();
+    }
+  }
+
+  @Watch('paymentId')
+  @Watch('authToken')
+  @Watch('paymentService')
```

Comment:
I think since `paymentId`, `authToken`, and `paymentService` are not props or pieces of state this will never fire. However, I don't think it's necessary to watch for any changes here anyways. `payment-details.tsx` watches for `paymentId` and `authToken` changes then re-initializes and updates `getPaymentDetails` which is in state. That should cause a re-render, which should cause this component to be re-initialized